// @kriegcc/react-cookie-consent/dist/react-cookie-consent.es.js
var we = Object.defineProperty;
var be = (r, e, t) => e in r ? we(r, e, { enumerable: !0, configurable: !0, writable: !0, value: t }) : r[e] = t;
var C = (r, e, t) => be(r, typeof e != "symbol" ? e + "" : e, t);
import Re, { createContext as J, useContext as V, useState as O, useEffect as se, useCallback as Q } from "react";
var A = { exports: {} }, y = {};
/**
 * @license React
 * react-jsx-runtime.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ee;
function xe() {
  if (ee) return y;
  ee = 1;
  var r = Symbol.for("react.transitional.element"), e = Symbol.for("react.fragment");
  function t(n, s, i) {
    var c = null;
    if (i !== void 0 && (c = "" + i), s.key !== void 0 && (c = "" + s.key), "key" in s) {
      i = {};
      for (var g in s)
        g !== "key" && (i[g] = s[g]);
    } else i = s;
    return s = i.ref, {
      $$typeof: r,
      type: n,
      key: c,
      ref: s !== void 0 ? s : null,
      props: i
    };
  }
  return y.Fragment = e, y.jsx = t, y.jsxs = t, y;
}
var _ = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var oe;
function ye() {
  return oe || (oe = 1, process.env.NODE_ENV !== "production" && function() {
    function r(o) {
      if (o == null) return null;
      if (typeof o == "function")
        return o.$$typeof === Se ? null : o.displayName || o.name || null;
      if (typeof o == "string") return o;
      switch (o) {
        case j:
          return "Fragment";
        case fe:
          return "Profiler";
        case ue:
          return "StrictMode";
        case ve:
          return "Suspense";
        case pe:
          return "SuspenseList";
        case he:
          return "Activity";
      }
      if (typeof o == "object")
        switch (typeof o.tag == "number" && console.error(
          "Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."
        ), o.$$typeof) {
          case h:
            return "Portal";
          case ge:
            return (o.displayName || "Context") + ".Provider";
          case de:
            return (o._context.displayName || "Context") + ".Consumer";
          case Ce:
            var a = o.render;
            return o = o.displayName, o || (o = a.displayName || a.name || "", o = o !== "" ? "ForwardRef(" + o + ")" : "ForwardRef"), o;
          case me:
            return a = o.displayName || null, a !== null ? a : r(o.type) || "Memo";
          case z:
            a = o._payload, o = o._init;
            try {
              return r(o(a));
            } catch {
            }
        }
      return null;
    }
    function e(o) {
      return "" + o;
    }
    function t(o) {
      try {
        e(o);
        var a = !1;
      } catch {
        a = !0;
      }
      if (a) {
        a = console;
        var l = a.error, f = typeof Symbol == "function" && Symbol.toStringTag && o[Symbol.toStringTag] || o.constructor.name || "Object";
        return l.call(
          a,
          "The provided key is an unsupported type %s. This value must be coerced to a string before using it here.",
          f
        ), e(o);
      }
    }
    function n(o) {
      if (o === j) return "<>";
      if (typeof o == "object" && o !== null && o.$$typeof === z)
        return "<...>";
      try {
        var a = r(o);
        return a ? "<" + a + ">" : "<...>";
      } catch {
        return "<...>";
      }
    }
    function s() {
      var o = N.A;
      return o === null ? null : o.getOwner();
    }
    function i() {
      return Error("react-stack-top-frame");
    }
    function c(o) {
      if (G.call(o, "key")) {
        var a = Object.getOwnPropertyDescriptor(o, "key").get;
        if (a && a.isReactWarning) return !1;
      }
      return o.key !== void 0;
    }
    function g(o, a) {
      function l() {
        B || (B = !0, console.error(
          "%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://react.dev/link/special-props)",
          a
        ));
      }
      l.isReactWarning = !0, Object.defineProperty(o, "key", {
        get: l,
        configurable: !0
      });
    }
    function p() {
      var o = r(this.type);
      return X[o] || (X[o] = !0, console.error(
        "Accessing element.ref was removed in React 19. ref is now a regular prop. It will be removed from the JSX Element type in a future release."
      )), o = this.props.ref, o !== void 0 ? o : null;
    }
    function R(o, a, l, f, E, m, F, L) {
      return l = m.ref, o = {
        $$typeof: u,
        type: o,
        key: a,
        props: m,
        _owner: E
      }, (l !== void 0 ? l : null) !== null ? Object.defineProperty(o, "ref", {
        enumerable: !1,
        get: p
      }) : Object.defineProperty(o, "ref", { enumerable: !1, value: null }), o._store = {}, Object.defineProperty(o._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: 0
      }), Object.defineProperty(o, "_debugInfo", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: null
      }), Object.defineProperty(o, "_debugStack", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: F
      }), Object.defineProperty(o, "_debugTask", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: L
      }), Object.freeze && (Object.freeze(o.props), Object.freeze(o)), o;
    }
    function k(o, a, l, f, E, m, F, L) {
      var d = a.children;
      if (d !== void 0)
        if (f)
          if (ke(d)) {
            for (f = 0; f < d.length; f++)
              x(d[f]);
            Object.freeze && Object.freeze(d);
          } else
            console.error(
              "React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead."
            );
        else x(d);
      if (G.call(a, "key")) {
        d = r(o);
        var b = Object.keys(a).filter(function(Ee) {
          return Ee !== "key";
        });
        f = 0 < b.length ? "{key: someKey, " + b.join(": ..., ") + ": ...}" : "{key: someKey}", Z[d + f] || (b = 0 < b.length ? "{" + b.join(": ..., ") + ": ...}" : "{}", console.error(
          `A props object containing a "key" prop is being spread into JSX:
  let props = %s;
  <%s {...props} />
React keys must be passed directly to JSX without using spread:
  let props = %s;
  <%s key={someKey} {...props} />`,
          f,
          d,
          b,
          d
        ), Z[d + f] = !0);
      }
      if (d = null, l !== void 0 && (t(l), d = "" + l), c(a) && (t(a.key), d = "" + a.key), "key" in a) {
        l = {};
        for (var $ in a)
          $ !== "key" && (l[$] = a[$]);
      } else l = a;
      return d && g(
        l,
        typeof o == "function" ? o.displayName || o.name || "Unknown" : o
      ), R(
        o,
        d,
        m,
        E,
        s(),
        l,
        F,
        L
      );
    }
    function x(o) {
      typeof o == "object" && o !== null && o.$$typeof === u && o._store && (o._store.validated = 1);
    }
    var w = Re, u = Symbol.for("react.transitional.element"), h = Symbol.for("react.portal"), j = Symbol.for("react.fragment"), ue = Symbol.for("react.strict_mode"), fe = Symbol.for("react.profiler"), de = Symbol.for("react.consumer"), ge = Symbol.for("react.context"), Ce = Symbol.for("react.forward_ref"), ve = Symbol.for("react.suspense"), pe = Symbol.for("react.suspense_list"), me = Symbol.for("react.memo"), z = Symbol.for("react.lazy"), he = Symbol.for("react.activity"), Se = Symbol.for("react.client.reference"), N = w.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE, G = Object.prototype.hasOwnProperty, ke = Array.isArray, D = console.createTask ? console.createTask : function() {
      return null;
    };
    w = {
      "react-stack-bottom-frame": function(o) {
        return o();
      }
    };
    var B, X = {}, K = w["react-stack-bottom-frame"].bind(
      w,
      i
    )(), H = D(n(i)), Z = {};
    _.Fragment = j, _.jsx = function(o, a, l, f, E) {
      var m = 1e4 > N.recentlyCreatedOwnerStacks++;
      return k(
        o,
        a,
        l,
        !1,
        f,
        E,
        m ? Error("react-stack-top-frame") : K,
        m ? D(n(o)) : H
      );
    }, _.jsxs = function(o, a, l, f, E) {
      var m = 1e4 > N.recentlyCreatedOwnerStacks++;
      return k(
        o,
        a,
        l,
        !0,
        f,
        E,
        m ? Error("react-stack-top-frame") : K,
        m ? D(n(o)) : H
      );
    };
  }()), _;
}
var te;
function _e() {
  return te || (te = 1, process.env.NODE_ENV === "production" ? A.exports = xe() : A.exports = ye()), A.exports;
}
var I = _e();
const ae = J(void 0), Te = () => {
  const r = V(ae);
  if (r === void 0)
    throw new Error("useCookieConsent must be used within a CookieConsentProvider");
  return r;
}, ie = J(
  void 0
);
/*! js-cookie v3.0.5 | MIT */
function M(r) {
  for (var e = 1; e < arguments.length; e++) {
    var t = arguments[e];
    for (var n in t)
      r[n] = t[n];
  }
  return r;
}
var Ie = {
  read: function(r) {
    return r[0] === '"' && (r = r.slice(1, -1)), r.replace(/(%[\dA-F]{2})+/gi, decodeURIComponent);
  },
  write: function(r) {
    return encodeURIComponent(r).replace(
      /%(2[346BF]|3[AC-F]|40|5[BDE]|60|7[BCD])/g,
      decodeURIComponent
    );
  }
};
function U(r, e) {
  function t(s, i, c) {
    if (!(typeof document > "u")) {
      c = M({}, e, c), typeof c.expires == "number" && (c.expires = new Date(Date.now() + c.expires * 864e5)), c.expires && (c.expires = c.expires.toUTCString()), s = encodeURIComponent(s).replace(/%(2[346B]|5E|60|7C)/g, decodeURIComponent).replace(/[()]/g, escape);
      var g = "";
      for (var p in c)
        c[p] && (g += "; " + p, c[p] !== !0 && (g += "=" + c[p].split(";")[0]));
      return document.cookie = s + "=" + r.write(i, s) + g;
    }
  }
  function n(s) {
    if (!(typeof document > "u" || arguments.length && !s)) {
      for (var i = document.cookie ? document.cookie.split("; ") : [], c = {}, g = 0; g < i.length; g++) {
        var p = i[g].split("="), R = p.slice(1).join("=");
        try {
          var k = decodeURIComponent(p[0]);
          if (c[k] = r.read(R, k), s === k)
            break;
        } catch {
        }
      }
      return s ? c[s] : c;
    }
  }
  return Object.create(
    {
      set: t,
      get: n,
      remove: function(s, i) {
        t(
          s,
          "",
          M({}, i, {
            expires: -1
          })
        );
      },
      withAttributes: function(s) {
        return U(this.converter, M({}, this.attributes, s));
      },
      withConverter: function(s) {
        return U(M({}, this.converter, s), this.attributes);
      }
    },
    {
      attributes: { value: Object.freeze(e) },
      converter: { value: Object.freeze(r) }
    }
  );
}
var v = U(Ie, { path: "/" });
class Oe {
  constructor(e) {
    C(this, "consentStorageManager");
    this.consentStorageManager = e;
  }
  setLocalStorageItem(e, t) {
    if (!this.consentStorageManager.hasConsentFor(e, "LocalStorage")) {
      console.warn(`Cannot store item to local storage. No consent is given for key ${e}.`);
      return;
    }
    localStorage.setItem(e, t);
  }
  getLocalStorageItem(e) {
    return this.consentStorageManager.hasConsentFor(e, "LocalStorage") ? localStorage.getItem(e) : (console.warn(`Cannot get item from local storage. No consent is given for key ${e}.`), null);
  }
  removeLocalStorageItem(e) {
    if (!this.consentStorageManager.hasConsentFor(e, "LocalStorage")) {
      console.warn(`Cannot remove item from local storage. No consent is given for key ${e}.`);
      return;
    }
    localStorage.removeItem(e);
  }
  setSessionStorageItem(e, t) {
    if (!this.consentStorageManager.hasConsentFor(e, "SessionStorage")) {
      console.warn(`Cannot store item to session storage. No consent is given for key ${e}.`);
      return;
    }
    sessionStorage.setItem(e, t);
  }
  getSessionStorageItem(e) {
    return this.consentStorageManager.hasConsentFor(e, "SessionStorage") ? sessionStorage.getItem(e) : (console.warn(`Cannot get item from session storage. No consent is given for key ${e}.`), null);
  }
  removeSessionStorageItem(e) {
    if (!this.consentStorageManager.hasConsentFor(e, "SessionStorage")) {
      console.warn(`Cannot remove item from session storage. No consent is given for key ${e}.`);
      return;
    }
    sessionStorage.removeItem(e);
  }
  setCookie(e, t, n) {
    if (!this.consentStorageManager.hasConsentFor(e, "Cookie")) {
      console.warn(`Cannot store cookie. No consent is given for key ${e}.`);
      return;
    }
    v.set(e, t, n);
  }
  getCookie(e) {
    if (!this.consentStorageManager.hasConsentFor(e, "Cookie"))
      return console.warn(`Cannot get cookie. No consent is given for key ${e}.`), null;
    const t = v.get(e);
    return t === void 0 ? null : t;
  }
  removeCookie(e, t) {
    if (!this.consentStorageManager.hasConsentFor(e, "Cookie")) {
      console.warn(`Cannot remove cookie. No consent is given for key ${e}.`);
      return;
    }
    v.remove(e, t);
  }
}
class Ae {
  constructor(e) {
    C(this, "consentStatus");
    this.consentStatus = e;
  }
  hasConsentFor(e, t) {
    const n = this.consentStatus.get(e);
    return !!(n !== void 0 && n.hasConsent && n.location === t);
  }
  isKeyKnown(e, t) {
    const n = this.consentStatus.get(e);
    return n !== void 0 && n.location === t;
  }
  updateConsentStatus(e) {
    this.consentStatus = e, this.handleConsentRevocation();
  }
  handleConsentRevocation() {
    this.consentStatus.forEach((e, t) => {
      e.hasConsent || this.deleteKeyInLocation(t, e.location);
    });
  }
  deleteKeyInLocation(e, t) {
    switch (t) {
      case "LocalStorage":
        localStorage.removeItem(e);
        return;
      case "SessionStorage":
        sessionStorage.removeItem(e);
        return;
      case "Cookie":
        v.remove(e);
        return;
      default:
        throw new Error(`Unknown storage location "${t}".`);
    }
  }
}
const Me = ({ children: r }) => {
  const { cookieCategories: e } = Te(), [t] = O(
    new Ae(re(e))
  ), [n] = O(new Oe(t));
  return se(() => {
    t.updateConsentStatus(re(e));
  }, [t, e]), /* @__PURE__ */ I.jsx(ie.Provider, { value: { consentStorage: n }, children: r });
};
function re(r) {
  const e = /* @__PURE__ */ new Map();
  return r.forEach((t) => {
    const n = t.cookies, s = t.isEnabled;
    n.forEach((i) => {
      e.set(i.key, { location: i.location, hasConsent: s });
    });
  }), e;
}
const ce = J(
  void 0
), Pe = ({ children: r }) => {
  const [e, t] = O(!1), n = Q(() => {
    t(!0);
  }, []), s = Q(() => {
    t(!1);
  }, []);
  return /* @__PURE__ */ I.jsx(ce.Provider, { value: { isModalOpen: e, openModal: n, closeModal: s }, children: r });
}, S = "cookie_consent_preferences", P = 30 * 24 * 60 * 60 * 1e3;
function je() {
  return le() !== void 0;
}
function le() {
  const r = v.get(S);
  if (r !== void 0)
    try {
      return JSON.parse(r).consentDetails;
    } catch (e) {
      console.error("Error parsing cookie consent preferences:", e);
      return;
    }
}
function Ne(r) {
  const e = v.get(S);
  let t;
  const n = {
    date: new Date(Date.now()).toISOString(),
    choice: r
  };
  if (e)
    try {
      t = JSON.parse(e), t.consentDetails = n;
    } catch (i) {
      console.error("Error parsing existing cookie, creating a new one:", i), t = {
        consensualCategoryIds: [],
        expires: new Date(Date.now() + P).toISOString(),
        consentDetails: n
      };
    }
  else
    t = {
      consensualCategoryIds: [],
      expires: new Date(Date.now() + P).toISOString(),
      consentDetails: n
    };
  const s = JSON.stringify(t);
  v.set(S, s, {
    sameSite: "strict",
    expires: new Date(t.expires)
  });
}
function De() {
  Fe() && v.remove(S);
}
function Fe() {
  return v.get(S) !== void 0;
}
function Le() {
  const r = v.get(S);
  if (r === void 0)
    return [];
  try {
    return JSON.parse(r).consensualCategoryIds || [];
  } catch (e) {
    return console.error("Error parsing cookie consent preferences:", e), [];
  }
}
function Y(r) {
  const e = v.get(S);
  let t;
  if (e)
    try {
      t = JSON.parse(e), t.consensualCategoryIds = r;
    } catch (s) {
      console.error("Error parsing existing cookie, creating a new one:", s), t = {
        consensualCategoryIds: r,
        expires: new Date(Date.now() + P).toISOString(),
        consentDetails: void 0
      };
    }
  else
    t = {
      consensualCategoryIds: r,
      expires: new Date(Date.now() + P).toISOString(),
      consentDetails: void 0
    };
  const n = JSON.stringify(t);
  v.set(S, n, {
    sameSite: "strict",
    expires: new Date(t.expires)
  });
}
function $e() {
  const r = v.get(S);
  if (r !== void 0)
    try {
      return JSON.parse(r).expires;
    } catch (e) {
      console.error("Error parsing cookie consent preferences:", e);
      return;
    }
}
const We = ({
  initialCookieCategories: r,
  children: e
}) => {
  const t = ne(r), [n, s] = O(
    t
  ), [i, c] = O(!je());
  se(() => {
    const u = ne(r);
    return s(u), () => {
      Y(
        T(u)
      );
    };
  }, [r]);
  const g = (u) => {
    const h = n.get(u);
    h && (h.toggleIsEnabled(), Y(T(n)), s(new Map(n)));
  }, p = (u) => T(n).includes(u), R = (u) => {
    switch (u) {
      case "ACCEPT_ALL":
        n.forEach((h) => h.enable());
        break;
      case "REJECT_ALL":
        n.forEach((h) => {
          h.isMandatory || h.disable();
        });
        break;
    }
    Y(T(n)), s(new Map(n)), Ne(u), c(!1);
  }, k = () => {
    n.forEach((u) => {
      u.isMandatory || u.disable();
    }), s(new Map(n)), De(), c(!0);
  }, x = () => {
    c(!0);
  }, w = () => {
    const u = le();
    if (u !== void 0)
      return {
        choice: u.choice,
        choiceDate: u.date,
        enabledCategories: T(n),
        cookieExpirationDate: $e() || ""
      };
  };
  return /* @__PURE__ */ I.jsx(
    ae.Provider,
    {
      value: {
        cookieCategories: n,
        toggleCategoryIsEnable: g,
        isCookieCategoryIdConsensual: p,
        isConsentRequired: i,
        askForConsent: x,
        giveConsent: R,
        rejectConsent: k,
        cookieConsentDetails: w()
      },
      children: /* @__PURE__ */ I.jsx(Me, { children: /* @__PURE__ */ I.jsx(Pe, { children: e }) })
    }
  );
};
function T(r) {
  return Array.from(r.values()).filter((e) => e.isEnabled).map((e) => e.id);
}
function ne(r) {
  const e = Le();
  return new Map(
    Array.from(r.entries()).map(([t, n]) => {
      const s = n.clone();
      return e.includes(t) ? s.enable() : s.disable(), [t, s];
    })
  );
}
const qe = () => {
  const r = V(ie);
  if (r === void 0)
    throw new Error("useConsentStorage must be used within a ConsentStorageProvider");
  return r;
};
class W {
  constructor(e, t, n, s) {
    C(this, "key");
    C(this, "description");
    C(this, "location");
    C(this, "duration");
    this.key = e, this.description = t, this.location = n, this.duration = s;
  }
  clone() {
    return new W(this.key, this.description, this.location, this.duration);
  }
}
class q {
  constructor(e, t, n, s = !1) {
    C(this, "id");
    C(this, "title");
    C(this, "description");
    C(this, "isEnabled");
    C(this, "isMandatory");
    C(this, "cookies");
    this.id = e, this.title = t, this.description = n, this.isMandatory = s, this.isEnabled = s, this.cookies = [];
  }
  addCookie(e) {
    this.cookies.push(e);
  }
  toggleIsEnabled() {
    if (this.isMandatory) {
      console.warn("This cookie category is mandatory. It cannot be disabled.");
      return;
    }
    this.isEnabled = !this.isEnabled;
  }
  enable() {
    this.isEnabled = !0;
  }
  disable() {
    if (this.isMandatory) {
      console.warn("This cookie category is mandatory. It cannot be disabled.");
      return;
    }
    this.isEnabled = !1;
  }
  clone() {
    const e = new q(
      this.id,
      this.title,
      this.description,
      this.isMandatory
    );
    return e.isEnabled = this.isEnabled, e.cookies = this.cookies.map((t) => t.clone()), e;
  }
}
function Ye(r) {
  return ["necessary", "functional", "analytics", "advertisement"].includes(r);
}
function Ue(r) {
  return ["Cookie", "LocalStorage", "SessionStorage", "Cache"].includes(r);
}
function ze(r) {
  if (!Array.isArray(r.categories))
    return !1;
  for (const e of r.categories) {
    if (!Ye(e.id) || typeof e.title != "string" || typeof e.description != "string" || typeof e.isMandatory != "boolean" || !Array.isArray(e.cookies))
      return !1;
    for (const t of e.cookies)
      if (typeof t.key != "string" || typeof t.description != "string" || !Ue(t.location) || typeof t.duration != "string")
        return !1;
  }
  return !0;
}
function Ge(r) {
  const e = /* @__PURE__ */ new Map();
  return r.categories.map((t) => {
    const n = new q(
      t.id,
      t.title,
      t.description,
      t.isMandatory
    );
    t.cookies.forEach((s) => {
      const i = new W(
        s.key,
        s.description,
        s.location,
        s.duration
      );
      n.addCookie(i);
    }), e.set(n.id, n);
  }), e;
}
const Be = () => {
  const r = V(ce);
  if (r === void 0)
    throw new Error("useCookieConsentModal must be used within a CookieConsentModalProvider");
  return r;
};
export {
  Oe as ConsentStorage,
  W as Cookie,
  q as CookieCategory,
  We as CookieConsentProvider,
  Ge as loadCookies,
  qe as useConsentStorage,
  Te as useCookieConsent,
  Be as useCookieConsentModal,
  ze as validateCookiesJsonFile
};
